### ISSUE TITLE
Make sure there is no attempt to re-attach `DETACHING` channel after connection becomes `SUSPENDED`

### ISSUE THREAD

**Issue by lawrence-forooghian** [ably/specification#242]
See [internal conversation](https://ably-real-time.slack.com/archives/C8SPU4589/p1732272857922879?thread_ts=1732191641.458019&cid=C8SPU4589).

We think the spec does not define what should happen when the following sequence of events occurs:

1. the user calls `detach()`, putting the channel into `DETACHING`;
2. some time after, the connection becomes `SUSPENDED`;
3. some time after, the connection becomes `CONNECTED` again.

In particular, does the channel reattach and start receiving messages again? The desired behaviour is that it not do so. The way in which to make this happen is probably to make sure that the channel ends up in the `DETACHED` state when the connection becomes `SUSPENDED`.

The aims of this task are to:

- [ ] make sure that this behaviour is specified
- [ ] once specified, create SDK repo issues to implement this
- [ ] in particular, implement in JS as a priority

**Comment by sacOO7** [comment]
Yeah, I agree, if connection goes suspended and existing channel state is detaching, then it should go detached. This also goes as per https://sdk.ably.com/builds/ably/specification/main/features/#RTL5j
Mainly, [Connection state change side effects spec](https://sdk.ably.com/builds/ably/specification/main/features/#RTL3) might need revision based on given channel state.
